JniBridge.bee.cs

```diff
@@ -0,0 +1,379 @@
+        "::android::app::AlertDialog_Builder",
+        "::android::app::NotificationManager",
+        "::android::app::Presentation",
+        "::android::content::Context",
+        "::android::graphics::Color",
+        "::android::graphics::ImageFormat",
+        "::android::graphics::drawable::ColorDrawable",
+        "::android::hardware::display::DisplayManager",
+        "::android::hardware::Camera",
+        "::android::hardware::input::InputManager",
+        "::android::hardware::GeomagneticField",
+        "::android::location::LocationManager",
+        "::android::media::AudioAttributes_Builder",
+        "::android::media::AudioFocusRequest_Builder",
+        "::android::media::AudioManager",
+        "::android::media::MediaCodec",
+        "::android::media::MediaCodec::BufferInfo",
+        "::android::media::MediaExtractor",
+        "::android::media::MediaFormat",
+        "::android::media::MediaRouter",
+        "::android::net::ConnectivityManager",
+        "::android::net::wifi::WifiManager",
+        "::android::os::Build",
+        "::android::os::Build_VERSION",
+        "::android::os::HandlerThread",
+        "::android::os::Environment",
+        "::android::os::PowerManager",
+        "::android::os::Process",
+        "::android::os::Vibrator",
+        "::android::provider::Settings_Secure",
+        "::android::provider::Settings_System",
+        "::android::telephony::TelephonyManager",
+        "::android::telephony::SubscriptionManager",
+        "::android::telephony::SubscriptionInfo",
+        "::android::view::Choreographer",
+        "::android::view::Display",
+        "::android::view::Gravity",
+        "::android::view::SurfaceView",
+        "::android::view::WindowManager",
+        "::android::webkit::MimeTypeMap",
+        "::android::widget::CheckBox",
+        "::android::widget::CompoundButton_OnCheckedChangeListener",
+        "::android::widget::ProgressBar",
+        "::java::lang::Character",
+        "::java::lang::System",
+        "::java::lang::SecurityException",
+        "::java::lang::NoSuchMethodError",
+        "::java::lang::ClassCastException",
+        "::java::lang::UnsatisfiedLinkError",
+        "::java::io::FileNotFoundException",
+        "::java::net::HttpURLConnection",
+        "::java::nio::channels::Channels",
+        "::java::util::HashSet",
+        "::java::util::Map_Entry",
+        "::java::util::NoSuchElementException",
+        "::java::util::Scanner",
+        "::java::util::zip::ZipFile",
+        "::javax::net::ssl::X509TrustManager",
+        "::javax::net::ssl::TrustManagerFactory",
+        "::java::security::KeyStore",
+        "::com::google::android::gms::ads::identifier::AdvertisingIdClient",
+        "::com::google::android::gms::common::GooglePlayServicesAvailabilityException",
+        "::com::google::android::gms::common::GooglePlayServicesNotAvailableException",
+    };
+
+    private static readonly string[] kMacOSClasses = new[]
```

Comment:
Curious about this one, why are there Mac specific classes, if it would be Windows, would there be a different array ?

If not, can we remove MacOS mentions here and in related places.

Feels strange, that there's Mac specific thing here, if there's a reason, please add comment here

Reply:
It's comes from the fact that perl/make build code only had android and osx support, the later for tests. Actually, even now it doesn't build on Windows (a guess, but I think the include directories are OSX specific).

Reply:
Help me to understand here - kMacOSClasses is used to generate some cpp files ? which are used for producing jnibridge static library, which is used by Android player (not Mac player).

Reply:
> Help me to understand here - kMacOSClasses is used to generate some cpp files ? which are used for producing jnibridge static library, which is used by Android player (not Mac player)? So why anything Mac specific is used for Android source code generation?

We generate code and build a separate MacOS version of JNI bridge that is used by test program.
The setup is:
- for Mac we generate code for all different codegen cases and run a test program that tries them out.
- for Android we generate code for Android APIs that we use from C++ code.

Reply:
ok